### PR TITLE
Case insensitive queries

### LINKED ISSUE #295: Support case insensitive queries
Requested by a user. We should probably come up with a syntax better than the [c] used in predicates. Thoughts @bdash @kristiandupont @jpsim @cmelchior ?

### PR DESCRIPTION
**Pull request by alazier**
Closes #295

### CODE REVIEW

File: src/object-store/src/parser/parser.cpp [hunk 1]
```diff
@@ -66,8 +66,9 @@ struct argument : seq< one< '$' >, must< argument_index > > {};
 
 // expressions and operators
 struct expr : sor< dq_string, sq_string, number, argument, true_value, false_value, key_path > {};
+struct case_insensitive : pegtl_istring_t("[c]"){};
```
Comment (appden) [inline src/object-store/src/parser/parser.cpp]: Only lowercase `[c]` should be allowed, like `NSPredicate`

Comment (appden) [inline src/object-store/src/parser/parser.cpp]: Nvm, they actually do allow `[C]` :smile:

File: src/object-store/tests/query.json [hunk 2]
```diff
@@ -197,10 +197,10 @@
...
         ["QueryThrows", "StringObject", "stringCol CONTAINS $0", 1],
 
-        ["Disabled", "QueryCount", 3, "StringObject", "stringCol ==[c] 'a'"],
-        ["Disabled", "QueryCount", 5, "StringObject", "stringCol BEGINSWITH[c] 'A'"],
-        ["Disabled", "QueryCount", 4, "StringObject", "stringCol ENDSWITH[c] 'c'"],
-        ["Disabled", "QueryCount", 2, "StringObject", "stringCol CONTAINS[c] 'B'"]
+        ["QueryCount", 3, "StringObject", "stringCol ==[c] 'a'"],
+        ["QueryCount", 5, "StringObject", "stringCol BEGINSWITH[c] 'A'"],
+        ["QueryCount", 4, "StringObject", "stringCol ENDSWITH[c] 'c'"],
+        ["QueryCount", 2, "StringObject", "stringCol CONTAINS[c] 'B'"]
```
Comment (appden) [inline src/object-store/tests/query.json]: I believe the existing code allows `[c]` on non-string comparisons and just ignores it. I think we really should be throwing an error in that case, and test here that we do.

Comment (alazier) [inline src/object-store/tests/query.json]: This only applied to the `==` operator as the other operators throw for non-string properties. I don't think we need to thrown an exception in this case.

Comment (appden) [inline src/object-store/tests/query.json]: Ok. I think for correctness sake we should throw in this case, but that's up to you.

**Comment by appden** [comment]
Looks good to merge :+1:

**Pull request merged**